src/transformers/models/auto/tokenization_auto.py

```diff
@@ -348,9 +348,57 @@ class AutoTokenizer:
     def __init__(self):
         raise EnvironmentError(
             "AutoTokenizer is designed to be instantiated "
-            "using the `AutoTokenizer.from_pretrained(pretrained_model_name_or_path)` method."
+            "using the `AutoTokenizer.from_pretrained(pretrained_model_name_or_path)` or `AutoTokenizer.from_model_name(config)` method."
         )
 
+    @classmethod
+    @replace_list_option_in_docstrings(TOKENIZER_MAPPING_NAMES)
+    def from_model_name(cls, model_name, *args, **kwargs):
+        r"""
+        Instantiate one of the tokenizer classes of the library by passing the required vocabulary file.
+
+        The tokenizer class to instantiate is selected based on the :obj:`model_name` which is passed as an argument.
+
+        List options
+
+        Params:
+            model_name (:obj:`str`):
+                The :obj:`model_name` associated to the tokenizer class that should be instantiated. Should be one of
+                the keys shown in bold above.
+            use_fast (:obj:`bool`, `optional`, defaults to :obj:`True`):
+                Whether or not to try to load the fast version of the tokenizer.
+            args (additional positional arguments, `optional`):
+                Will be passed to the Tokenizer ``__init__()`` method. Can be used to pass the required vocabulary
+                files such as ``vocab_file`` or ``merges_file``.
+            kwargs (additional keyword arguments, `optional`):
+                Will be passed to the Tokenizer ``__init__()`` method. Can be used to pass the required vocabulary
+                files such as ``vocab_file=/path/tol/vocab_file.json`` and/or ``merges_file=/path/to/merges_file.txt``
+                as well as to set special tokens like ``bos_token``, ``eos_token``, ``unk_token``, ``sep_token``,
+                ``pad_token``, ``cls_token``, ``mask_token``, ``additional_special_tokens``. See parameters in the
+                ``__init__()`` for more details.
+
+        Examples::
+
+            >>> from transformers import AutoTokenizer
+
+            >>> # Instantiate BERT-like tokenizer
+            >>> tokenizer = AutoTokenizer.from_model_name("bert", vocab_file="./vocab.txt")
+
+            >>> # Instantiate GPT2-like tokenizer
+            >>> tokenizer = AutoTokenizer.from_model_name("gpt2", vocab_file="./vocab.json", merges_file="./merges.txt")
+        """
+
+        use_fast = kwargs.pop("use_fast", True)
+
+        tokenizer_class_name, tokenizer_class_name_fast = TOKENIZER_MAPPING_NAMES[model_name]
+
+        if use_fast and tokenizer_class_name_fast is not None:
```

Comment:
A small suggestion here, wouldn't we return an error message when the user explicitly requests a slow or fast version and it doesn't exist for the requested type?

Reply:
Think a warning might be a good idea! In `AutoTokenizer.from_pretrained(..., use_fast=True)` we always fall back to the slow version if fast doesn't exist, so I think we should keep the same design here cc @sgugger @LysandreJik WDYT?

Reply:
Yes, since `use_fast=True` is the default, the user usually did not explicitly request the tokenizer, so there should not be a warning (they scare users).

Another design could be to have the parameter default to None, and when it's None do the behavior we have right now (try fast then slow). When it's an explicit boolean, we could then either issue a warning or an error message. An error message would probably be nicer and would allow us to get rid of some `assert tokenizer.is_fast` in the examples.

In all cases, we should have the same design in AutoTokenizer class methods obviously.

Wdyt?

Reply:
Think it'd be a good idea to default to `None` and raise an warning/error if the boolean is explicitly set to `True` and there is no fast tokenizer. I think I'd opt for a warning though (maybe saying that this behavior will result in an error in the future) as it would be breaking for `AutoTokenizer.from_pretrained(...)`